scipy/ndimage/tests/test_filters.py

```diff
@@ -86,6 +86,33 @@ def test_valid_origins():
         assert_raises(ValueError, filter, data, 3, origin=2)
 
 
+def test_multiple_modes():
+    # Test that the filters with multiple mode cababilities for different
+    # dimensions give the same result as applying a single mode.
+    arr = np.zeros((100, 100), float)
+    arr[50, 50] = 1
+    arr[50, 0] = 1
+    arr[0, 0] = 1
+
+    mode1 = 'reflect'
+    mode2 = ['reflect', 'reflect']
```

Comment:
The problem with two dimensions is that the array has no sides, only corners, which get extrapolation information from more than one axis averaged. This is why I would prefer at least 3x3, which has pure side pixels, which can show the different behavior explicitly.